File: swift-bridge/Sources/CoreLocationBridge/Visit.swift
import CoreLocation
import Foundation

func cl_visit_object(_ visit: CLVisit) -> [String: Any] {
    [
        "arrival_date": visit.arrivalDate.timeIntervalSince1970,
        "departure_date": visit.departureDate.timeIntervalSince1970,
        "coordinate": cl_coordinate_object(visit.coordinate),
        "horizontal_accuracy": visit.horizontalAccuracy,
    ]
}

File: src/visit.rs
use serde::{Deserialize, Serialize};

use crate::location::Coordinate;

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
/// Snapshot of `CLVisit`.
pub struct Visit {
    /// Matches `CLVisit.arrivalDate`.
    pub arrival_date: f64,
    /// Matches `CLVisit.departureDate`.
    pub departure_date: f64,
    /// Matches `CLVisit.coordinate`.
    pub coordinate: Coordinate,
    /// Matches `CLVisit.horizontalAccuracy`.
    pub horizontal_accuracy: f64,
}
